### DISCUSSION TITLE
Create signed download/upload URL

### DISCUSSION THREAD

**Discussion by bernisnukic** [drakkan/sftpgo#569] [Ideas]
Would it be possible to generate signed url's that can be shared to non authenticated users? For example we can generate a signed download url for a specific file with a 1 time use. Or create a signed upload url where someone could post data and it would be one time use (or timed).

Reference:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/ShareObjectPreSignedURL.html
https://docs.aws.amazon.com/AmazonS3/latest/userguide/PresignedUrlUploadObject.html

**Comment by drakkan** [comment]
Hi,

something similar is planned. 

I had never thought about an upload URL, however I would like to add a sharing feature which allows you to select one or more files / folders and generate a download URL (the client will download the selected files/folders as zip). Maybe I can also add the number of  times the download is allowed and maybe an optional password.

I don't have a definite schedule a sponsorship could help

**Reply by drakkan** [reply]
Hi,

this is now supported

https://github.com/drakkan/sftpgo/commit/3bc58f5988fcfef970fd080a7d73716488f253ab

testing is welcome, thanks

**Reply by bernisnukic** [reply]
Hello, the new features are working great, thank you!

I was wondering about something else, is it possible to recursively create a folder when you specify an upload path via the rest api? Or if you specify a path with the folder creation rest api endpoint?

**Reply by drakkan** [reply]
If you mean for REST API different from public shares I think this could be done. 

For uploads to public shares this should not be allowed or at least it should be configurable and the default should remain as it is now (cannot upload to subdirectories)

**Reply by bernisnukic** [reply]
Yeah I mean the REST API, unrelated to shares.
